### PR TITLE
Feature/refactor code to typescript

### LINKED ISSUE #22: Create global logger based on winston for example available which is available the StorefrontApiContext
## What is the motivation for adding/enhancing this feature?

Currently on most when not all places we use console.log/error/warn for logging. I would thing that a global logger instance based on like winston or an other logging library would improve the over all logging for development and production. 

## What are the acceptance criteria 

- [ ] Global logger instance
- [ ] Log Level configurable in the config file and via environment variable

## Can you complete this feature request by yourself?

- [x] YES
- [ ] NO

### PR DESCRIPTION
**Pull request by ResuBaka**
### Related issues

closes #22 

### Short description and why it's useful

* graphql example type annotation
* resolvers are loaded as typescript or javascript file
* added db and config to graphql context
* DbContext and initializeDb has now better annotation
* Fixed dist output
  * compiles scripts now too
  * the folder structure has changed a bit in the dist folder -> pm2 config was updated for it
* Added logger
  * Missing for it is documentation how to add you own instance to it, so you can use any logger you need
* Updated example module a but with the new types

### Screenshots of visual changes before/after (if there are any)

**IMPORTANT NOTICE** - Remember to update `CHANGELOG.md` with a description of your change

### Contribution and currently important rules acceptance

- [x] I read and followed [contribution rules](https://github.com/DivanteLtd/storefront-api/blob/develop/CONTRIBUTING.md)

### CODE REVIEW

**Comment by ResuBaka** [comment]
I still want to change more code to typescript and add better type hinting.

But when you guys want this in the source code asap I can move it to **Ready for review**.

**Comment by pkarw** [comment]
@ResuBaka take your time, that's perfect

File: src/lib/elastic.ts [hunk 1]
```diff
@@ -61,37 +65,39 @@ function adjustQuery (esQuery, entityType, config) {
...
+    esConfig.auth = {
+      username: config.get<string>('elasticsearch.user'),
+      password: config.get<string>('elasticsearch.password')
+    }
   }
   return new es.Client(esConfig)
 }
 
-function putAlias (db, originalName, aliasName, next) {
+export function putAlias (db: Client, originalName: string, aliasName: string, next) {
```
Comment (ResuBaka) [inline src/lib/elastic.ts]: Here I need your help.

What type should next be?

Comment (pkarw) [reply src/lib/elastic.ts]: it's a callable - function kind of :)

Comment (ResuBaka) [reply src/lib/elastic.ts]: I thing I have found for all callbacks the right type for it.

**Comment by ResuBaka** [comment]
It is now ready to review and we should have one that makes a QA on it so we still now that all works like we expect.

We need a followup for the logger class that we add the missing documentation and @pkarw can you look if it is usable or if there are missing helper/functions for it?

File: src/modules/default-vsf/api/extensions/mailchimp-subscribe/index.js [hunk 2]
```diff
@@ -1,4 +1,4 @@
-import { apiStatus } from 'src/lib/util'
+import { apiStatus } from '../../../../../lib/util'
```
Comment (pkarw) [inline src/modules/default-vsf/api/extensions/mailchimp-subscribe/index.js]: Why this change?
